compat/src/index.d.ts

```diff
@@ -0,0 +1,71 @@
+import * as hooks from '../../hooks';
+import { VNode, PreactElement } from '../../src/internal';
+import { Component, createContext, createRef, Fragment, toChildArray, ComponentChildren, ComponentChild } from '../../src';
+import { FunctionalComponent } from 'preact';
+import { ForwardFn } from './internal';
+
+export { Component, createContext, createRef, Fragment } from '../../src';
+export * from '../../hooks';
+
+export const version: string;
+
+export function createPortal (vnode: VNode, container: PreactElement): VNode<any>;
+
+export function createElement (type: VNode["type"], props?: any, Children?: ComponentChildren): VNode<any>;
+
+export function cloneElement (element: VNode<any>): VNode<any>;
+
+export function render (vnode: VNode<any>, parent: PreactElement, callback?: () => void): Component | null;
+
+export function unmountComponentAtNode (container: Element | Document | ShadowRoot | DocumentFragment): boolean;
+
+export function createFactory (type: VNode["type"]): VNode<any>;
+
+export function isValidElement (element: any): boolean;
+
+export function findDOMNode (component: Component): PreactElement | null;
+
+export class PureComponent extends Component<P, S> {
+  public isPureReactComponenet: boolean;
+
+  public shouldComponentUpdate (props: P, state: S): boolean;
+}
+
+export function memo (c: FunctionalComponent, comparer: (prev: any, next: any) => boolean): FunctionalComponent;
```

Comment:
`comparer` should be optional and`memo` is used to compare `props`. We should add that into our types:

```ts
export function memo<P = {}>(component: FunctionalComponent<P>, comparer?: (prev: P, next: P) => boolean): FunctionComponent<P>;
```